tools/get_tcwind_csv.py

```diff
@@ -35,37 +36,39 @@ def convert_open_water_1minute_to_10minute(df):
     return df
 
 
-def _get_hazard(all_lats, all_lons, location_ids=None,
-                terrain_correction='full_terrain_gust',
-                wind_speed_averaging_period='3_seconds',
-                product='deepcyc', scenario='current_climate',
-                time_horizon='now', return_period=None):
-
-    assert len(all_lats) == len(all_lons), 'Mismatching number of lats and lons'
-    if location_ids is not None:
-        assert len(all_lats) == len(location_ids)
 
-    num_calls = ceil(len(all_lats) / 100)
-    if return_period is None and product.lower() == 'deepcyc':
-        # We are pulling the full stochastic history - do one lat, lon pair at a time.
-        num_calls = len(all_lats)
+def _do_queries_serially(all_lats, all_lons,
+                         terrain_correction,
+                         wind_speed_averaging_period,
+                         product, scenario,
+                         time_horizon, return_period):
 
     if product.lower() == 'deepcyc':
         m = DeepCyc()
     else:
         m = Metryc()
 
-    convert_to_10_minute = False
-    if wind_speed_averaging_period == '10_minutes':
-        convert_to_10_minute = True
-        wind_speed_averaging_period = '1_minute'
-        assert terrain_correction == 'open_water', \
-            '10 minutes wind speed averaging period only supported for Open Water terrain type.'
+    num_calls = ceil(len(all_lats) / 100)
+    if return_period is None and product.lower() == 'deepcyc':
+        # We are pulling the full stochastic history - do one lat, lon pair at a time.
+        num_calls = len(all_lats)
 
-    df = None
+    token_age = 0
+    start_time = time.time()
+    all_query_dfs = []
     for lats, lons in zip(np.array_split(all_lats, num_calls),
                           np.array_split(all_lons, num_calls)):
 
+        token_age = time.time() - start_time
```

Comment:
you may not need to have `token_age`. it can simply `if time.time() - start_time > 3600:`